### DISCUSSION TITLE
Need help with Audi MIB2

### DISCUSSION THREAD

**Discussion by Darreman** [Mr-MIBonk/M.I.B._More-Incredible-Bash#51] [Q&A] [answered]
Hello.

Im hoping to get help to recover my mib2 if possible. I installed M.I.B and probably did some bad stuff in there because my system now boots and get stuck on Radio screen. The car buttons are unresponsive only thing that works is reset. After some time 1-3 minutes the unit reboots. I can enter mmx via dlink adapter but not rcc.  I can enter RCC with uart serial usb board.
I made backup before doing anything i M.I.B but im unsure how or if it can be used to fix my problem at this stage.
Any tips highly appreciated

Here is the backup log

Logging_MU1438-MHI2_ER_AUG22_K3346-JTB-07427.05.1611730A8E:
HW Number: 052 - PN: 8W0035036C
Variant: FM2-P-TNL-EU-AU-MLE
EEPROM Variant: MU-P-LN-EU
Dataset Version: 8W0909545J
Model ID: 019402 - Car Model: Audi A4 (8W0)
VIN -> CAR:  - MIB: 

 ls --> total 2919
      64 drwxrwxrwx 11 root      root          32768 Oct 24 00:05 .
      64 drwxrwxrwx 11 root      root          32768 Oct 24 00:05 ..
      64 drwxrwxrwx  2 root      root          32768 Oct 24 00:22 .System Volume Information
       2 -rwxrwxrwx  1 root      root            986 Oct 27 13:43 .git-blame-ignore-revs
       1 -rwxrwxrwx  1 root      root            399 Oct 27 13:43 .gitattributes
      64 drwxrwxrwx  2 root      root          32768 Oct 27 13:43 .github
       1 -rwxrwxrwx  1 root      root            132 Oct 27 13:43 .gitignore
    1703 -rwxrwxrwx  1 root      root         871810 Oct 27 13:43 2022_06_MHI2_Password_List_V4.0.pdf
      81 -rwxrwxrwx  1 root      root          41255 Oct 27 13:43 GEM.png
      30 -rwxrwxrwx  1 root      root          14939 Oct 27 13:43 LICENSE
      64 drwxrwxrwx  4 root      root          32768 Oct 27 13:43 Launcher
       1 -rwxrwxrwx  1 root      root             60 Oct 27 13:43 M.I.B. - More Incredible Bash - Mibsolution Wiki.url
       0 -rwxrwxrwx  1 root      root              0 Oct 27 13:43 NOT FOR COMMERCIAL USE - IF YOU BOUGHT THIS YOU GOT RIPPED OFF.txt
     284 -rwxrwxrwx  1 root      root         144962 Oct 27 13:43 PATCH COMPATIBILITY TABLE.pdf
      13 -rwxrwxrwx  1 root      root           6469 Oct 27 13:43 README.md
       1 -rwxrwxrwx  1 root      root             21 Oct 27 13:43 VERSION
       1 -rwxrwxrwx  1 root      root            296 Jan 01  1980 WMPInfo.xml
       0 -rwxrwxrwx  1 root      root              0 Oct 27 13:43 _Swdlautorun.txt
      64 drwxrwxrwx  3 root      root          32768 Oct 27 13:43 apps
      64 drwxrwxrwx  4 root      root          32768 Jan 01  1980 backup
      77 -rwxrwxrwx  1 root      root          39351 Oct 27 13:43 bash.png
      64 drwxrwxrwx  2 root      root          32768 Oct 27 13:43 config
      64 drwxrwxrwx  3 root      root          32768 Oct 27 13:43 esd
       4 -rwxrwxrwx  1 root      root           1995 Oct 27 13:43 metainfo2.txt
      64 drwxrwxrwx 10 root      root          32768 Oct 27 13:43 mod
      64 drwxrwxrwx  4 root      root          32768 Oct 28 06:04 patches
      16 -rwxrwxrwx  1 root      root           7989 Oct 27 13:43 start

RCC-backup---->
ifs-root-part2 (ASCII): 00f0
3 image type=2(IFS     )  ba0000...1a89977 (15636856)|93...213 (120):LOCKED
0x00ba0000 - 7899EE00

Found JTB-07427.05.1611730A8E Backup, will check if all complete...
Version info..... -> done :-)
FEC folders...... -> done :-)
Shadow files..... -> done :-)
Dataset's........ -> done :-)
EEPROM........... -> done :-)
storage1......... -> done :-)
storage2......... -> done :-)
backing up backup -> part 1 done
ls -als /net/mmx/mnt/boardbook --> total 192
      64 drwxrwxrwx  3 root      root          32768 Jan 01  1970 .
      64 drwxrwxr-x  5 root      root          32768 Jan 01  1970 ..
      64 drwxrwxrwx  3 root      root          32768 Jan 01  1970 MU1438
Free space on /net/mmx/mnt/boardbook 900M

RCC-launcher---->
-- _Swdlautorun.txt found
-- autostart already disabled
-- Skip MOST enabled
-- Illegal and withdrawn FECs removed
-- Cleanup done!
AppDevelopment.jar found
-- GEM version OK - no update needed
-- M.I.B GEM link installed
-- Development Mode - GEM - activated

RCC-backup---->
ifs-root-part2 (ASCII): 00f0
3 image type=2(IFS     )  ba0000...1a89977 (15636856)|93...213 (120):LOCKED
0x00ba0000 - 7899EE00

Found JTB-07427.05.1611730A8E Backup, will check if all complete...
Partition info... -> done :-)
SWDL............. -> done :-)
Version info..... -> Backup is already there!
GEM info........ -> done :-)
Folder lists..... -> done :-)
FEC folders...... -> Backup is already there!
HMI folders...... -> done :-)
Shadow files..... -> Backup is already there!
lsd.jxe.......... -> done :-)
Dataset's........ -> Backup is already there!
EEPROM........... -> Backup is already there!
storage1......... -> Backup is already there!
storage2......... -> Backup is already there!
RCC Flashlock.... -> done :-)
RCC.............. -> done :-)
MMX.............. -> done :-)
IFS-stage2.......4512+0 records in
4512+0 records out
 -> done :-)
OFFSET's......... -> done :-)
backing up backup -> Backup part 1 is already there!
ls -als /net/mmx/mnt/boardbook --> total 192
      64 drwxrwxrwx  3 root      root          32768 Jan 01  1970 .
      64 drwxrwxr-x  5 root      root          32768 Jan 01  1970 ..
      64 drwxrwxrwx  3 root      root          32768 Jan 01  1970 MU1438
Free space on /net/mmx/mnt/boardbook 772M

MMX-fecel---->

add FECs to FEC file from M.I.B backup folder
Existing FecContainer parsed
FEC File build done!

copy FEC file from /patches/MHI2_ER_AUG22_K3346_MU1438_PATCH... -> done :-)

MMX-fecel---->
copy EL file from /patches/MHI2_ER_AUG22_K3346_MU1438_PATCH... -> done :-)

RCC-carp---->
CarPlay and Android Auto activated now

RCC-flash---->
ifs-root-part2 (ASCII): 00f0
3 image type=2(IFS     )  ba0000...1a89977 (15636856)|93...213 (120):LOCKED
0x00ba0000 - 7899EE00

Found JTB-07427.05.1611730A8E Backup, will check if all complete...
Partition info... -> Backup is already there!
SWDL............. -> Backup is already there!
Version info..... -> Backup is already there!
GEM info........ -> done :-)
Folder lists..... -> Backup is already there!
FEC folders...... -> Backup is already there!
HMI folders...... -> Backup is already there!
Shadow files..... -> Backup is already there!
lsd.jxe.......... -> Backup is already there!
Dataset's........ -> Backup is already there!
EEPROM........... -> Backup is already there!
storage1......... -> Backup is already there!
storage2......... -> Backup is already there!
RCC Flashlock.... -> Backup is already there!
RCC.............. -> Backup is already there!
MMX.............. -> Backup is already there!
IFS-stage2....... -> Backup is already there!
OFFSET's......... -> Backup is already there!
backing up backup -> Backup part 1 is already there! -> Backup part 2 is already there!

ls -als /net/mmx/mnt/boardbook --> total 192
      64 drwxrwxrwx  3 root      root          32768 Jan 01  1970 .
      64 drwxrwxr-x  5 root      root          32768 Jan 01  1970 ..
      64 drwxrwxrwx  3 root      root          32768 Jan 01  1970 MU1438
Free space on /net/mmx/mnt/boardbook 772M

search File MU1438-ifs-root-part2-0x00ba0000-7899EE00.ifs

[flashlock] unlocking flash: DONEFlashing on OFFSET 0x00ba0000...
Label: PL_qnx_flashit_10495A QNX: RL_qnx_650_10494A Built: 2010/12/10_11-21-47_UTC
/net/rcc/dev/fs0
--------
Flash size: 0x04000000 [= 512 x 0x20000]
Base addr:  0x00ba0000
File size:  0x00ee9988 (15636872)
End addr:   0x01a89988
*** erasing:     0x00BA0000 .. 0x01A89987 [01A9FFFF] ........................................................................................................................ 
*** programming: 0x00BA0000 .. 0x01A89987            ..................................................................................

Best regards
David

**Comment by harman-f** [comment]
You flash process was interrupted.

Can you still log into RCC via piutty and D-Link?

**Reply by Darreman** [reply]
Hey mate

When I try to login to rcc with dlink it says connection refused. But I can enter rcc with serial connection, and get to emergency flash utility and so on
Thanks

best regards
David

**Comment by harman-f** [comment, marked as answer]
is MMX via telnet working?
If so, see if /net/rcc/ can still be accessed. 
Just a bit easier than UART.

To fix this you have to flash ifs-root-stage2 manually. It can be the patched one directly.
Most likely rcc will not be visible in /net/rcc/dev/fs0 in that case follow this:
https://mibwiki.one/doc/mount-rcc-fs0-6zJlPOu7Ml

**Reply by Darreman** [reply]
Thank you mate! I managed to flash ifs-root via uart  and unit is now working again
Thanks so much

best regards 
David
